packages/primitives/src/primitives/fetch.js

```diff
@@ -17,79 +17,33 @@ global.AbortSignal = AbortSignal
 // undici uses `process.nextTick`,
 // but process APIs doesn't exist in a runtime context.
 process.nextTick = setImmediate
-
-/**
- * A symbol used to store cookies in the headers module.
- */
-const SCookies = Symbol('set-cookie')
-
-/**
- * Patch HeadersList.append so that when a `set-cookie` header is appended
- * we keep it in an list to allow future retrieval of all values.
- */
-const __append = HeadersModule.HeadersList.prototype.append
-HeadersModule.HeadersList.prototype.append = function (name, value) {
-  const result = __append.call(this, name, value)
-  if (!this[SCookies]) {
-    Object.defineProperty(this, SCookies, {
-      configurable: false,
-      enumerable: false,
-      writable: true,
-      value: [],
-    })
+process.emitWarning = () => {}
+
+const Request = new Proxy(BaseRequest, {
+  construct(target, args) {
+    const [input, init] = args
+    return new target(input, addDuplexToInit(init))
+  },
+})
+
+const __entries = HeadersModule.Headers.prototype.entries
+HeadersModule.Headers.prototype.entries = function* () {
+  for (const [key, value] of __entries.call(this)) {
```

Comment:
question: Why do we need to wrap the original `entries`? Looks like this is for `set-cookie` but would like to know the background why undici's implementation is not sufficient for us.